### PR TITLE
Adds information about securing the Swagger endpoint

### LINKED ISSUE #1996: Generate Local Swagger Documentation Security Note Missing
## This issue is a:
- [ ] Bug on the website
- [x] New topic request
- [ ] Topic clarification request
- [ ] New DevDocs feature request
- [ ] Other
 
## Description:
The [current](https://devdocs.magento.com/guides/v2.2/rest/generate-local.html) documentation fails to mention that on a production system the `/swagger` path should be considered for mitigation (either via webserver configuration or M2 Admin URL Rewrite rule). Even though this path only lists the API endpoints available to anonymous users, it can also publish information about a Magento installation, such as it being Open Source vs Commerce ... or potentially show 3rd party extensions that publish API endpoints ... allowing a potential attacker to gain information about a target.

### Steps to reproduce
Simply go to a known M2 installation and update the path in the url to `/swagger`
  
### Expected result:
A suggestion in the documentation that on a production system, this path should be ignored ... or at least knowledge of the fact that without any action being taken, a list of anonymous API endpoints is available to anyone requesting the `/swagger` path.

## Possible solutions:
A simple note added to the page about a production consideration being to prevent access to this path, and adding a link to this note on a security page and production configuration page in the documentation. That way if someone is looking into suggested security configurations for M2 or if someone is looking at suggested production configurations for M2, that a link to this information is present.
 
## Additional information:
Basically without this issue being highlighted, many M2 installations have been, and will continue to be, installed leaving a vulnerability for information gathering from a potential attacker. If a potential attacker was trying to fingerprint a site or try to identify what the underlying framework is, at the very least the `/swagger` path will disclose if an installation is M2 Open Source or Commerce.

If this is something that is agreed to be taken into consideration, I am happy to go about creating a PR to publish a note about this and potential steps of mitigation.

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this pull request

This pull request (PR) adds info about disabling /swagger or using mod_rewrite

Fixes #1996 

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/rest/generate-local.html
https://devdocs.magento.com/guides/v2.3/rest/generate-local.html

whatsnew
Added a Security section to the [Generate a local REST reference](https://devdocs.magento.com/guides/v2.4/rest/generate-local.html) topic.

### CODE REVIEW

File: src/guides/v2.3/rest/generate-local.md [hunk 1]
```diff
@@ -63,7 +63,25 @@ To return the complete JSON schema, specify the `?services=all` parameter in the
 
 The base URL for returning the asynchronous schema is `http://<magento_host>/rest/<store_code>/async/schema`.
 
+## Security
+
+By default, the `/swagger` endpoint is open on Magneto instances. Leaving it open on a public instance could potentially expose system information that should not be readable.
```
Comment (BarnyShergold) [inline src/guides/v2.3/rest/generate-local.md:68]:
```suggestion
By default, the `/swagger` endpoint is open on Magento instances. Leaving it open on a public instance could potentially expose system information that should not be readable. 
```

File: src/guides/v2.4/rest/generate-local.md [hunk 2]
```diff
@@ -60,7 +60,25 @@ To return the complete JSON schema, specify the `?services=all` parameter in the
 
 The base URL for returning the asynchronous schema is `http://<magento_host>/rest/<store_code>/async/schema`.
 
+## Security
+
+By default, the `/swagger` endpoint is open on Magneto instances. Leaving it open on a public instance could potentially expose system information that should not be readable.
```
Comment (BarnyShergold) [inline src/guides/v2.4/rest/generate-local.md:65]:
```suggestion
By default, the `/swagger` endpoint is open on Magento instances. Leaving it open on a public instance could potentially expose system information that should not be readable. 
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great addition

File: src/guides/v2.4/rest/generate-local.md (hunk 2 above)
Comment (dobooth) [reply src/guides/v2.4/rest/generate-local.md:65]: Of all the words to misspell, it's the name of my employer.

**Review by keharper — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**